examples/numba_support.py

```diff
@@ -0,0 +1,973 @@
+    def __init__(self):
+        super().__init__(name="UInt3")
+
+
+uint3 = UInt3()
+
+
+# UInt3 data model
+
+
+@register_model(UInt3)
+class UInt3Model(models.StructModel):
+    def __init__(self, dmm, fe_type):
+        members = [
+            ("x", types.uint32),
+            ("y", types.uint32),
+            ("z", types.uint32),
+        ]
+        super().__init__(dmm, fe_type, members)
+
+
+make_attribute_wrapper(UInt3, "x", "x")
+make_attribute_wrapper(UInt3, "y", "y")
+make_attribute_wrapper(UInt3, "z", "z")
+
+
+# Prototype a function to construct a uint3
+
+
+def make_uint3(x, y, z):
+    pass
+
+
+@register
+class MakeUInt3(ConcreteTemplate):
+    key = make_uint3
+    cases = [signature(uint3, types.uint32, types.uint32, types.uint32)]
+
+
+register_global(make_uint3, types.Function(MakeUInt3))
+
+
+# make_uint3 lowering
+
+
+@lower(make_uint3, types.uint32, types.uint32, types.uint32)
+def lower_make_uint3(context, builder, sig, args):
+    # u4 = uint32
+    u4_3 = cgutils.create_struct_proxy(uint3)(context, builder)
+    u4_3.x = args[0]
+    u4_3.y = args[1]
+    u4_3.z = args[2]
+    return u4_3._getvalue()
+
+
+# Temporary Payload Parameter Pack
+class PayloadPack(types.Type):
+    def __init__(self):
+        super().__init__(name="PayloadPack")
+
+
+payload_pack = PayloadPack()
+
+
+# UInt3 data model
+
+
+@register_model(PayloadPack)
+class PayloadPackModel(models.StructModel):
+    def __init__(self, dmm, fe_type):
+        members = [
+            ("p0", types.uint32),
+            ("p1", types.uint32),
+            ("p2", types.uint32),
+        ]
+        super().__init__(dmm, fe_type, members)
+
+
+make_attribute_wrapper(PayloadPack, "p0", "p0")
+make_attribute_wrapper(PayloadPack, "p1", "p1")
+make_attribute_wrapper(PayloadPack, "p2", "p2")
+
+# OptiX typedefs and enums
+# -----------
+
+OptixVisibilityMask = types.Integer("OptixVisibilityMask", bitwidth=32, signed=False)
+OptixTraversableHandle = types.Integer(
+    "OptixTraversableHandle", bitwidth=64, signed=False
+)
+
+
+OPTIX_RAY_FLAG_NONE = 0
+# class OptixRayFlags(Enum):
+#     OPTIX_RAY_FLAG_NONE = 0
+#     OPTIX_RAY_FLAG_DISABLE_ANYHIT = 1 << 0
+#     OPTIX_RAY_FLAG_ENFORCE_ANYHIT = 1 << 1
+#     OPTIX_RAY_FLAG_TERMINATE_ON_FIRST_HIT = 1 << 2
+#     OPTIX_RAY_FLAG_DISABLE_CLOSESTHIT = 1 << 3,
+#     OPTIX_RAY_FLAG_CULL_BACK_FACING_TRIANGLES = 1 << 4
+#     OPTIX_RAY_FLAG_CULL_FRONT_FACING_TRIANGLES = 1 << 5
+#     OPTIX_RAY_FLAG_CULL_DISABLED_ANYHIT = 1 << 6
+#     OPTIX_RAY_FLAG_CULL_ENFORCED_ANYHIT = 1 << 7
+
+
+# OptiX types
+# -----------
+
+# Typing for OptiX types
+
+
+class SbtDataPointer(types.RawPointer):
+    def __init__(self):
+        super().__init__(name="SbtDataPointer")
+
+
+sbt_data_pointer = SbtDataPointer()
+
+
+# Models for OptiX types
+
+
+@register_model(SbtDataPointer)
+class SbtDataPointerModel(models.OpaqueModel):
+    pass
+
+
+# Params
+# ------------
+
+# Structures as declared in triangle.h
+
+
+class ParamsStruct:
```

Comment:
`Params`, `MissData` belong to the user defined code space, ideally we should create some useful helper function that allows user to custom define their structure.